lib/puppet/parser/compiler.rb

```diff
@@ -865,11 +865,32 @@ def create_settings_scope
     scope = @topscope.class_scope(settings_type)
 
     env = environment
+    settings_hash = {}
     Puppet.settings.each do |name, setting|
       next if name == :name
-      scope[name.to_s] = env[name]
+      name = name.to_s
+      # Construct a hash (in anticipation it will be set in top scope under a name like $settings)
+      settings_hash[name] = transform_setting(env[name])
```

Comment:
ok, it is needed as a string key in the settings_hash and in scope, will lookup with the symbol in env